tephra-core/src/main/java/org/apache/tephra/TransactionManager.java

```diff
@@ -917,6 +933,15 @@ private void doCommit(long transactionId, long writePointer, Set<ChangeId> chang
         invalidArray = invalid.toLongArray();
         LOG.info("Tx invalid list: removed committed tx {}", transactionId);
       }
+    } else {
+      long[] checkpointPointers = previous.getCheckpointWritePointers().toLongArray();
+      if (checkpointPointers != null && checkpointPointers.length > 0) {
+        // adjust the write pointer to be the last checkpoint of the tx and remove all checkpoints from inProgress
+        writePointer = checkpointPointers[checkpointPointers.length - 1];
+        for (long checkpointPointer : checkpointPointers) {
+          inProgress.remove(checkpointPointer);
```

Comment:
the `LongArrayList` returned by `getCheckpointWritePointers()` method is actually an instance of `Collection<Long>` as well, so you can just call `inProgress.keySet().removeAll(previous.getCheckpointWritePointers())`